### PR TITLE
Security workflow for C

### PR DESCRIPTION
**Pull request by fochoao-alt**
##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+Use this section to tell people about which versions of your project are
+currently being supported with security updates.
+
+| Version | Supported          |
+| ------- | ------------------ |
+| 5.1.x   | :white_check_mark: |
+| 5.0.x   | :x:                |
+| 4.0.x   | :white_check_mark: |
+| < 4.0   | :x:                |
```
Comment (kroeckx) [inline SECURITY.md:13]: This is clearly wrong. This file should probably just be removed, everything in this file is documented elsewhere.

**Comment by kroeckx** [comment]
I can confirm that the file is the same as the one you get when using the advanced setup as described in https://docs.github.com/en/code-security/code-scanning/creating-an-advanced-setup-for-code-scanning/configuring-advanced-setup-for-code-scanning

**Comment by kroeckx** [comment]
So it doesn't know the build system, and I guess it needs manual setup.

**Comment by fochoao-alt** [comment]
> So it doesn't know the build system, and I guess it needs manual setup.

Fixing it ASAP.

File: SECURITY.md (hunk 1 above)
Comment (fochoao-alt) [reply SECURITY.md]: yes that one is wrong, but for the other it is not.

**Comment by fochoao-alt** [comment]
> So it doesn't know the build system, and I guess it needs manual setup.

flawfinder.yml fixes it.

**Review by fochoao-alt — COMMENTED** [review]
Changed this workflow for cross compatibility.

File: .github/workflows/flawfinder.yml [hunk 2]
```diff
@@ -0,0 +1,291 @@
+name: flawfinder
+
+on: [pull_request, push]
+
+permissions:
+  contents: read, write
+
```
Comment (fochoao-alt) [inline .github/workflows/flawfinder.yml]:
```suggestion
    actions: read
    contents: read
    security-events: write
```

File: .github/workflows/flawfinder.yml [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+    output: 'flawfinder_results.sarif'
+  - name: Upload analysis results to GitHub Security tab
+    uses: github/codeql-action/upload-sarif@v3
+    with:
+      sarif_file: ${{github.workspace}}/flawfinder_results.sarif
+  cross-compilation:
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: [
```
Comment (fochoao-alt) [inline .github/workflows/flawfinder.yml]:
```suggestion
  cross-compilation:
  strategy:
  fail-fast: false
  matrix:
  platform: [
```

File: .github/workflows/flawfinder.yml [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+            target: linux-armv4,
+            tests: -test_includes -test_store -test_x509_store
+          }, {
+            arch: arm-linux-gnueabihf,
+            libs: libc6-dev-armhf-cross,
+            target: linux-armv4,
+            tests: -test_includes -test_store -test_x509_store
+          }, {
+            # gcc hppa seems to have some potential compiler issues
+            # with -O2 on this platform, reduce optimization to -01
```
Comment (fochoao-alt) [inline .github/workflows/flawfinder.yml:52]:
```suggestion

```

File: .github/workflows/flawfinder.yml [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+          }, {
+            arch: sh4-linux-gnu,
+            libs: libc6-dev-sh4-cross,
+            target: no-async linux-latomic,
+            tests: -test_includes -test_store -test_x509_store
+          },
+
+          # These build with shared libraries but they crash when run
+          # They mirror static builds above in order to cover more of the
+          # code base.
```
Comment (fochoao-alt) [inline .github/workflows/flawfinder.yml:104]:
```suggestion

```

File: .github/workflows/flawfinder.yml [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+          name: "cross-compiles@${{ matrix.platform.arch }}"
+          path: artifacts.tar.gz
+          if-no-files-found: ignore
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
```
Comment (fochoao-alt) [inline .github/workflows/flawfinder.yml:291]:
```suggestion
        runs-on: ubuntu-latest
        steps:
        - name: install package repository
          if: matrix.platform.ppa != ''
          run: |
            sudo add-apt-repository ppa:${{ matrix.platform.ppa }}
        - name: install packages
          run: |
            sudo apt-get update
            sudo apt-get -yq --force-yes install \
                gcc-${{ matrix.platform.arch }} \
                ${{ matrix.platform.libs }}
        - uses: actions/checkout@v4
        - name: checkout fuzz/corpora submodule
          run: git submodule update --init --depth 1 fuzz/corpora
    
        - name: config with FIPS
          if: matrix.platform.fips != 'no'
          run: |
            ./config --banner=Configured --strict-warnings enable-fips \
                     --cross-compile-prefix=${{ matrix.platform.arch }}- \
                     ${{ matrix.platform.target }}
        - name: config without FIPS
          if: matrix.platform.fips == 'no'
          run: |
            ./config --banner=Configured --strict-warnings \
                     --cross-compile-prefix=${{ matrix.platform.arch }}- \
                     ${{ matrix.platform.target }}
        - name: config dump
          run: ./configdata.pm --dump
    
        - name: make
          run: make -s -j4
    
        - name: install qemu
          if: matrix.platform.tests != 'none'
          run: sudo apt-get -yq --force-yes install qemu-user
    
        - name: Set QEMU environment
          if: matrix.platform.qemucpu != ''
          run: echo "QEMU_CPU=${{ matrix.platform.qemucpu }}" >> $GITHUB_ENV
    
        - name: Set OpenSSL caps environment
          if: matrix.platform.opensslcapsname != ''
          run: echo "OPENSSL_${{ matrix.platform.opensslcapsname }}=\
                     ${{ matrix.platform.opensslcaps }}" >> $GITHUB_ENV
    
        - name: get cpu info
          run: cat /proc/cpuinfo
    
        - name: make all tests
          if: github.event_name == 'push' && matrix.platform.tests == ''
          run: |
            .github/workflows/make-test \
                      TESTS="-test_afalg" \
                      QEMU_LD_PREFIX=/usr/${{ matrix.platform.arch }}
        - name: make some tests
          if: github.event_name == 'push' && matrix.platform.tests != 'none' && matrix.platform.tests != ''
          run: |
            .github/workflows/make-test \
                      TESTS="${{ matrix.platform.tests }} -test_afalg" \
                      QEMU_LD_PREFIX=/usr/${{ matrix.platform.arch }}
        - name: make evp tests
          if: github.event_name == 'pull_request' && matrix.platform.tests != 'none'
          run: |
            .github/workflows/make-test \
                      TESTS="test_evp*" \
                      QEMU_LD_PREFIX=/usr/${{ matrix.platform.arch }}
        - name: save artifacts
          uses: actions/upload-artifact@v4
          with:
            name: "cross-compiles@${{ matrix.platform.arch }}"
            path: artifacts.tar.gz
            if-no-files-found: ignore
```

File: .github/workflows/flawfinder.yml (hunk 2 above)
Comment (fochoao-alt) [inline .github/workflows/flawfinder.yml]:
```suggestion
    actions: read
    contents: read
    security-events: write
```

File: .github/workflows/flawfinder.yml (hunk 3 above)
Comment (fochoao-alt) [inline .github/workflows/flawfinder.yml]:
```suggestion
  cross-compilation:
  strategy:
  fail-fast: false
  matrix:
  platform: [
```

File: .github/workflows/flawfinder.yml (hunk 4 above)
Comment (fochoao-alt) [inline .github/workflows/flawfinder.yml:52]:
```suggestion

```

File: .github/workflows/flawfinder.yml (hunk 5 above)
Comment (fochoao-alt) [inline .github/workflows/flawfinder.yml:104]:
```suggestion

```

File: .github/workflows/flawfinder.yml (hunk 6 above)
Comment (fochoao-alt) [inline .github/workflows/flawfinder.yml:291]:
```suggestion
        runs-on: ubuntu-latest
        steps:
        - name: install package repository
          if: matrix.platform.ppa != ''
          run: |
            sudo add-apt-repository ppa:${{ matrix.platform.ppa }}
        - name: install packages
          run: |
            sudo apt-get update
            sudo apt-get -yq --force-yes install \
                gcc-${{ matrix.platform.arch }} \
                ${{ matrix.platform.libs }}
        - uses: actions/checkout@v4
        - name: checkout fuzz/corpora submodule
          run: git submodule update --init --depth 1 fuzz/corpora
    
        - name: config with FIPS
          if: matrix.platform.fips != 'no'
          run: |
            ./config --banner=Configured --strict-warnings enable-fips \
                     --cross-compile-prefix=${{ matrix.platform.arch }}- \
                     ${{ matrix.platform.target }}
        - name: config without FIPS
          if: matrix.platform.fips == 'no'
          run: |
            ./config --banner=Configured --strict-warnings \
                     --cross-compile-prefix=${{ matrix.platform.arch }}- \
                     ${{ matrix.platform.target }}
        - name: config dump
          run: ./configdata.pm --dump
    
        - name: make
          run: make -s -j4
    
        - name: install qemu
          if: matrix.platform.tests != 'none'
          run: sudo apt-get -yq --force-yes install qemu-user
    
        - name: Set QEMU environment
          if: matrix.platform.qemucpu != ''
          run: echo "QEMU_CPU=${{ matrix.platform.qemucpu }}" >> $GITHUB_ENV
    
        - name: Set OpenSSL caps environment
          if: matrix.platform.opensslcapsname != ''
          run: echo "OPENSSL_${{ matrix.platform.opensslcapsname }}=\
                     ${{ matrix.platform.opensslcaps }}" >> $GITHUB_ENV
    
        - name: get cpu info
          run: cat /proc/cpuinfo
    
        - name: make all tests
          if: github.event_name == 'push' && matrix.platform.tests == ''
          run: |
            .github/workflows/make-test \
                      TESTS="-test_afalg" \
                      QEMU_LD_PREFIX=/usr/${{ matrix.platform.arch }}
        - name: make some tests
          if: github.event_name == 'push' && matrix.platform.tests != 'none' && matrix.platform.tests != ''
          run: |
            .github/workflows/make-test \
                      TESTS="${{ matrix.platform.tests }} -test_afalg" \
                      QEMU_LD_PREFIX=/usr/${{ matrix.platform.arch }}
        - name: make evp tests
          if: github.event_name == 'pull_request' && matrix.platform.tests != 'none'
          run: |
            .github/workflows/make-test \
                      TESTS="test_evp*" \
                      QEMU_LD_PREFIX=/usr/${{ matrix.platform.arch }}
        - name: save artifacts
          uses: actions/upload-artifact@v4
          with:
            name: "cross-compiles@${{ matrix.platform.arch }}"
            path: artifacts.tar.gz
            if-no-files-found: ignore
```

File: .github/workflows/flawfinder.yml [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+    - name: get cpu info
+      run: cat /proc/cpuinfo
+
+    - name: make all tests
+      if: github.event_name == 'push' && matrix.platform.tests == ''
+      run: |
+        .github/workflows/make-test \
+                  TESTS="-test_afalg" \
+                  QEMU_LD_PREFIX=/usr/${{ matrix.platform.arch }}
+    - name: make some tests
```
Comment (fochoao-alt) [inline .github/workflows/flawfinder.yml:191]:
```suggestion
        runs-on: ubuntu-latest
        steps:
        - name: install package repository
          if: matrix.platform.ppa != ''
          run: |
            sudo add-apt-repository ppa:${{ matrix.platform.ppa }}
        - name: install packages
          run: |
            sudo apt-get update
            sudo apt-get -yq --force-yes install \
                gcc-${{ matrix.platform.arch }} \
                ${{ matrix.platform.libs }}
        - uses: actions/checkout@v4
        - name: checkout fuzz/corpora submodule
          run: git submodule update --init --depth 1 fuzz/corpora
    
        - name: config with FIPS
          if: matrix.platform.fips != 'no'
          run: |
            ./config --banner=Configured --strict-warnings enable-fips \
                     --cross-compile-prefix=${{ matrix.platform.arch }}- \
                     ${{ matrix.platform.target }}
        - name: config without FIPS
          if: matrix.platform.fips == 'no'
          run: |
            ./config --banner=Configured --strict-warnings \
                     --cross-compile-prefix=${{ matrix.platform.arch }}- \
                     ${{ matrix.platform.target }}
        - name: config dump
          run: ./configdata.pm --dump
    
        - name: make
          run: make -s -j4
    
        - name: install qemu
          if: matrix.platform.tests != 'none'
          run: sudo apt-get -yq --force-yes install qemu-user
    
        - name: Set QEMU environment
          if: matrix.platform.qemucpu != ''
          run: echo "QEMU_CPU=${{ matrix.platform.qemucpu }}" >> $GITHUB_ENV
    
        - name: Set OpenSSL caps environment
          if: matrix.platform.opensslcapsname != ''
          run: echo "OPENSSL_${{ matrix.platform.opensslcapsname }}=\
                     ${{ matrix.platform.opensslcaps }}" >> $GITHUB_ENV
    
        - name: get cpu info
          run: cat /proc/cpuinfo
    
        - name: make all tests
          if: github.event_name == 'push' && matrix.platform.tests == ''
          run: |
            .github/workflows/make-test \
                      TESTS="-test_afalg" \
                      QEMU_LD_PREFIX=/usr/${{ matrix.platform.arch }}
        - name: make some tests
          if: github.event_name == 'push' && matrix.platform.tests != 'none' && matrix.platform.tests != ''
          run: |
            .github/workflows/make-test \
                      TESTS="${{ matrix.platform.tests }} -test_afalg" \
                      QEMU_LD_PREFIX=/usr/${{ matrix.platform.arch }}
        - name: make evp tests
          if: github.event_name == 'pull_request' && matrix.platform.tests != 'none'
          run: |
            .github/workflows/make-test \
                      TESTS="test_evp*" \
                      QEMU_LD_PREFIX=/usr/${{ matrix.platform.arch }}
        - name: save artifacts
          uses: actions/upload-artifact@v4
          with:
            name: "cross-compiles@${{ matrix.platform.arch }}"
            path: artifacts.tar.gz
            if-no-files-found: ignore
```

**Review by fochoao-alt — COMMENTED** [review]
Improved verbal communication

**Review by fochoao-alt — COMMENTED** [review]
Checking workflow.

**Review by fochoao-alt — COMMENTED** [review]
Checked.

**Review by fochoao-alt — COMMENTED** [review]
This one is sent to the designated address of email, which is included in the secrets area.

**Review by fochoao-alt — COMMENTED** [review]
Fixed flawfinder v2 in cross-compilaation mode, the another in simple mode, if successful sendfs You an email. And dependantbot.yml

**Comment by kroeckx** [comment]
So this is currently showing 5000 issues. I'm going to guess that more than 99% are false positives, and we'll probably not going to spend looking at all of them.

**Comment by nhorman** [comment]
so, looking at the docs for this new tool, it looks alot like a static analysis tool.  We already run nightly coverity checks, and have for quite some time.  Whats the advantage over coverity to adding this in as well?  Seems like a duplication of effort to me

**Comment by fochoao-alt** [comment]
> so, looking at the docs for this new tool, it looks alot like a static analysis tool. We already run nightly coverity checks, and have for quite some time. Whats the advantage over coverity to adding this in as well? Seems like a duplication of effort to me

Thanks for pointing that out, thanks for the information.

**Pull request closed without merging**